CONTRIBUTING.md

```diff
@@ -1,35 +1,152 @@
+
+## Building and setting up a development workspace
+
+TiKV is mostly written in Rust, but has components written in C++ (RocksDB) and Go (gRPC). We are currently using the Rust nightly toolchain. To guarantee consistency in coding style, we use linters and automated formatting tools. 
+
+### Prerequisites
+
+To build TiKV you'll need to at least have the following installed:
+
+* `git` - Version control
+* [`rustup`](https://rustup.rs/) - Rust installer and toolchain manager
+* `make` - Build tool (run common workflows)
+* `cmake` - Build tool (required for gRPC)
+* `awk` - Pattern scanning/processing language
+
+If you are targeting platforms other than x86_64 linux, you'll also need:
+
+* [`llvm` and `clang`](http://releases.llvm.org/download.html) - Used to generate bindings for different platforms and build native libraries (required for grpcio, rocksdb).
+
+### Getting the repository
+
+```
+git clone https://github.com/tikv/tikv.git
+cd tikv
+# Future instructions assume you are in this directory
+```
+
+### Configuring your Rust toolchain
+
+`rustup` is the official toolchain manager for Rust, similar to `rvm` or `rbenv` from the Ruby world.
+
+TiKV is pinned to a version of Rust using a `rust-toolchain` file. `rustup` and `cargo` will automatically use this file. We also use the `rustfmt` and `clippy` components, to install those:
+
+```bash
+rustup component add rustfmt
+rustup component add clippy
+```
+
+### Building and testing
+
+TiKV includes a `Makefile` with common workflows, you can also use `cargo`, as you would in many other Rust projects.
+
+At this point, you can build TiKV:
+
+```bash
+make build
+```
+
+During interactive development, you may prefer using `cargo check`, which will parse, borrow check, and lint your code, but not actually compile it:
+
+```bash
+cargo check --all
+```
+
+`cargo check` is particularly handy alongside `cargo-watch`, which runs a command each time you change a file.
+
+```bash
+cargo install cargo-watch
+cargo watch -s "cargo check -all"
+```
+
+When you're ready to test out your changes, use the `dev` task. It formats your codebase, builds with `clippy` enabled, and runs tests. This should run without failure before you create a PR. Unfortunately, some tests might fail intermittently and others don't pass on all platforms. If you're unsure, just ask!
+
+```bash
+make dev
+```
+
+You can run the test suite alone, or just run a specific test:
+
+```bash
+# Run the full suite
+make test
+# Run a specific test
+cargo test $TESTNAME
+```
+
+TiKV follows the Rust community coding style. We use Rustfmt and [Clippy](https://github.com/Manishearth/rust-clippy) to automatically format and lint our code. Using these tools is checked in our CI. These are included as a part of the `make dev` task.  You can also run them alone:
+
+```bash
+# Run Rustfmt
+cargo fmt --all
```

Comment:
Does this still cover everything due to #5289 ?

Reply:
Yes, in fact I don't think we even need the `--all`